internal/resources/metal/vlan/datasource.go

```diff
@@ -83,77 +84,53 @@ func dataSourceMetalVlanRead(d *schema.ResourceData, meta interface{}) error {
 	facilityRaw, facilityOk := d.GetOk("facility")
 
 	if !(vlanIdOk || (vxlanOk || projectOk || metroOk || facilityOk)) {
-		return equinix_errors.FriendlyError(fmt.Errorf("You must set either vlan_id or a combination of vxlan, project_id, and, metro or facility"))
+		return diag.Errorf("You must set either vlan_id or a combination of vxlan, project_id, and, metro or facility")
 	}
 
-	var vlan *packngo.VirtualNetwork
+	var vlan *metalv1.VirtualNetwork
 
 	if vlanIdOk {
 		var err error
-		vlan, _, err = client.ProjectVirtualNetworks.Get(
-			vlanIdRaw.(string),
-			&packngo.GetOptions{Includes: []string{"assigned_to"}},
-		)
+		vlan, _, err = client.VLANsApi.
+			GetVirtualNetwork(ctx, vlanIdRaw.(string)).
+			Include([]string{"assigned_to"}).Execute()
+
 		if err != nil {
-			return equinix_errors.FriendlyError(err)
+			return diag.FromErr(equinix_errors.FriendlyError(err))
 		}
 
 	} else {
 		projectID := projectRaw.(string)
 		vxlan := vxlanRaw.(int)
 		metro := metroRaw.(string)
 		facility := facilityRaw.(string)
-		vlans, _, err := client.ProjectVirtualNetworks.List(
-			projectRaw.(string),
-			&packngo.GetOptions{Includes: []string{"assigned_to"}},
-		)
+		vlans, _, err := client.VLANsApi.
+			FindVirtualNetworks(ctx, projectRaw.(string)).
+			Include([]string{"assigned_to"}).Execute()
+
 		if err != nil {
-			return equinix_errors.FriendlyError(err)
+			return diag.FromErr(equinix_errors.FriendlyError(err))
 		}
 
-		vlan, err = matchingVlan(vlans.VirtualNetworks, vxlan, projectID, facility, metro)
+		vlan, err = MatchingVlan(vlans.VirtualNetworks, vxlan, projectID, facility, metro)
 		if err != nil {
-			return equinix_errors.FriendlyError(err)
+			return diag.FromErr(equinix_errors.FriendlyError(err))
 		}
 	}
 
 	assignedDevices := []string{}
 	for _, d := range vlan.Instances {
-		assignedDevices = append(assignedDevices, d.ID)
+		assignedDevices = append(assignedDevices, d.GetId()) // instances is a list of href, should be list of device?
 	}
 
-	d.SetId(vlan.ID)
+	d.SetId(vlan.GetId())
 
-	return equinix_schema.SetMap(d, map[string]interface{}{
-		"vlan_id":     vlan.ID,
-		"project_id":  vlan.Project.ID,
-		"vxlan":       vlan.VXLAN,
-		"facility":    vlan.FacilityCode,
+	return diag.FromErr(equinix_schema.SetMap(d, map[string]interface{}{
+		"vlan_id":     vlan.GetId(),
+		"project_id":  vlan.AssignedTo.GetId(), // vlan assigned_to is an href; should be project?
+		"vxlan":       vlan.GetVxlan(),
+		"facility":    vlan.FacilityCode, // facility is deprecated, vlan is metro-scoped; remove this attr?
```

Comment:
In `metal-cli` we opted to [remove `facility` from VLAN subcommands because facility is deprecated and VLANs are at the metro level anyway;](https://github.com/equinix/metal-cli/pull/315/files#r1276358558) however I think removing it entirely could be a bigger impact in terraform so we might want to maintain support for it for now?

Reply:
Based on API responses I saw in Chrome developer tools, both `facility` and `facility_code` are `null` anyway, so maybe this is a non-issue?  Perhaps for now we just explicitly set it to null everywhere in code and deprecate the attribute in the resource & data source schemas so we can remove it later?